torch/csrc/jit/interpreter.cpp

```diff
@@ -904,14 +908,24 @@ struct InterpreterStateImpl {
         // std::cout << "executing " << pc << ": ";
         // function->dumpInstruction(std::cout, pc);
         // std::cout << "\n";
-        auto & inst = instructions[pc];
-        loadTensorsFromRegisters(inst.inputs, stack);
-        pc += 1 + inst.callback(stack);
-        for(int i = inst.outputs.size - 1; i >= 0; i--) {
-          int reg = get(inst.outputs,i);
-          registers[reg] = pop(stack);
-          // std::cout << "pop reg[" << reg << "];\n" << registers[reg].pImpl << "\n";
+        size_t new_pc;
+        try {
+          auto & inst = instructions[pc];
+          loadTensorsFromRegisters(inst.inputs, stack);
+          new_pc = pc + 1 + inst.callback(stack);
+          for(int i = inst.outputs.size - 1; i >= 0; i--) {
+            int reg = get(inst.outputs,i);
+            registers[reg] = pop(stack);
+            // std::cout << "pop reg[" << reg << "];\n" << registers[reg].pImpl << "\n";
+          }
+        } catch(std::exception & e) {
+          if(pc >= instructions.size() || !instructions[pc].debug_location)
+            throw; // rethrow original exception
```

Comment:
Ouch, why would this ever happen (the first condition)? It looks more like a segfault/memory corrupting, so the core really creates some false sense of safety. Maybe let's leave the second one only?

Reply:
yeah, I guess if the instruction is executing at all, then it has to be valid